osi_bank_reconciled_report/models/account_bank_reconciled_report.py

```diff
@@ -0,0 +1,490 @@
+
+    @api.model
+    def _get_amount(self, line, report_data):
+        if report_data["use_foreign_currency"]:
+            if line.currency_id == report_data["line_currency"]:
+                return line.amount_currency
+            else:
+                return line.currency_id._convert(
+                    line.amount_currency,
+                    report_data["line_currency"],
+                    self.env.user.company_id,
+                    line.date,
+                )
+        else:
+            return line.balance
+
+    @api.model
+    def _get_lines(self, options, line_id=None):
+        # Fetch data
+        report_data = self._get_bank_rec_report_data(options, line_id)
+        self = self.with_context(line_currency=report_data["line_currency"])
+        # Compute totals
+        unrec_tot = sum(
+            [
+                -(self._get_amount(aml, report_data))
+                for aml in report_data.get("reconciled_pmts", [])
+            ]
+        )
+        outstanding_plus_tot = sum(
+            [
+                st_line.amount
+                for st_line in report_data.get("reconciled_st_positive", [])
+            ]
+        )
+        outstanding_minus_tot = sum(
+            [
+                st_line.amount
+                for st_line in report_data.get("reconciled_st_negative", [])
+            ]
+        )
+        computed_stmt_balance = (
+            report_data["odoo_balance"]
+            + outstanding_plus_tot
+            + outstanding_minus_tot
+            + unrec_tot
+        )
+        difference = computed_stmt_balance - report_data["last_st_balance"]
+
+        # Build report
+        lines = []
+
+        lines.append(
+            self._add_title_line(
+                options,
+                _("Virtual GL Balance"),
+                amount=None
+                if self.env.user.company_id.totals_below_sections
+                else computed_stmt_balance,
+                level=0,
+            )
+        )
+
+        gl_title = _("Current balance of account %s")
+        if len(report_data["account_ids"]) > 1:
+            gl_title = _("Current balance of accounts %s")
+
+        accounts = self.env["account.account"].browse(report_data["account_ids"])
+        accounts_string = ", ".join(accounts.mapped("code"))
+        lines.append(
+            self._add_title_line(
+                options,
+                gl_title % accounts_string,
+                level=1,
+                amount=report_data["odoo_balance"],
+                date=options["date"]["date_from"],
+            )
+        )
+
+        lines.append(self._add_title_line(options, _("Operations to Process"), level=1))
+
+        if report_data.get("not_reconciled_st_positive") or report_data.get(
+            "not_reconciled_st_negative"
+        ):
+            lines.append(
+                self._add_title_line(
+                    options,
+                    _(
+                        "Uneconciled Bank \
```

Comment:
Misspelling, please change to Uneconciled to Unreconciled. v12 has an active PR to fix this which is why it wasn't part of the code here.

Reply:
Fixed. Thanks